include/swift/AST/DiagnosticsSema.def

```diff
@@ -1841,6 +1841,25 @@ NOTE(objc_implementation_requirement_here,none,
      "%kind0 declared in header here",
      (ValueDecl *))
 
+ERROR(objc_implementation_init_must_be_convenience, none,
+      "%kind0 is not valid in an '@_objcImplementation' extension because "
+      "Objective-C subclasses must be able to override "
+      "%select{designated|required}1 initializers",
+      (const ValueDecl *, /*isRequired=*/bool))
+NOTE(objc_implementation_init_turn_designated_to_convenience, none,
+     "add 'convenience' keyword to make this a convenience initializer",
+     ())
+NOTE(objc_implementation_init_turn_required_to_convenience, none,
+     "replace 'required' keyword with 'convenience' to make this a convenience "
+     "initializer",
+     ())
+
+// Fallback diagnostic; super-general by nature.
+ERROR(objc_implementation_member_requires_vtable, none,
+      "%kind0 is not valid in an '@_objcImplementation' extension because it "
```

Comment:
I take it from the comment that we want to avoid emitting this if possible, and in the test case that inspired this we emit a better, more actionable diagnostic (which is great!) I noticed that after that change, though, we no longer seem to have any test case that causes this diagnostic to be emitted. Is it possible to add one? I'm also wondering if there is a way we can word this diagnostic that's friendlier to folks who don't understand what a vtable is, or even if they do understand what one is, wouldn't know what to do to address the problem. I realize that may be heavily dependent on the member in question, though.

Reply:
If I knew of another way to cause this diagnostic, I would add another special case. Short of a debug flag that disables checking for all of the special cases, I don’t see a good way to test this.

I’m willing to consider wording the diagnostic differently. Maybe something like:

```
      "%kind0 is not valid in an '@_objcImplementation' extension because it "
      "is an overridable Swift-only %kindonly0",
```

Reply:
I've updated the wording, and also rebased on post-rebranch `main`.